stock_cancel/models/stock.py

```diff
@@ -46,16 +47,32 @@ def action_revert_done(self):
                     % (picking.name))
             if picking.invoice_id:
                 raise exceptions.Warning(
-                    _('Picking %s has invoices!') % (picking.name))
+                    _('Picking %s has been invoiced') % (picking.name))
+            for move in picking.move_lines:
+                quants = move.quant_ids.filtered(
```

Comment:
Why do you need to make this filter? With it, you're discarding associated quants that can be considered as a stopper for making the cancel, don't you think?